plugin/kubernetes/parse.go

```diff
@@ -57,13 +58,16 @@ func parseRequest(state request.Request) (r recordRequest, err error) {
 	if last < 0 {
 		return r, nil
 	}
-	r.podOrSvc = segs[last]
-	if r.podOrSvc != Pod && r.podOrSvc != Svc {
-		return r, errInvalidRequest
-	}
-	last--
-	if last < 0 {
-		return r, nil
+
+	if !external {
```

Comment:
why can't the caller check the return value of this function as-is and then decide if the request is usable. Why does this function need changing?

Reply:
Because the request format is different for external vs cluster names.

external names: service-name.namespace.zone.
cluster names: service-name.namespace.svc.zone.

Also, cluster names accept an endpoint segment for A records of service endpoints (pod ips). (endpoint.service-name.namespace.svc.zone).  External names do not (they only expose external IPs).